### ISSUE TITLE
Change state of request

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#452]
* As an analyst and ministry coordinator
* I want to a drop down menu
* so that I can I can change the state of a request

**Assumptions & Scope**	
What are the assumptions for this story?

What is in and not in scope?

**Acceptance Criteria**	

**Scenario 1: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 2: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 3: xxxxxx**
...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@arielleandrews, @JHarrietha-AOT, please review the above scenarios and advise on edits/omissions.

**Comment by arielleandrews** [comment]
Thanks Loren. For Scenario 2, the clock should update at/after midnight, rather than a 24-hour period. 

Should we include a Scenario 3 where after the due date has passed it turns red and counts down into the negative? Is there an action we should include in the design that will stop the day counter?

**Comment by lmullane** [comment]
@arielleandrews 

I'll update the scenario for a midnight change. Just confirming if we send a CFR to a ministry at 5pm on a business day, that when the clock strikes midnight, they are down to 9 days?

Yes, to a scenario 3.

**Comment by JHarrietha-AOT** [comment]
@arielleandrews I agree, I currently have the Item in the queue in red if it is past the IAO Records Due Date. Let me know if you think this is an effective way to show that in the queue. Was thinking about having it turn to orange/yellow when it is day before due date as well, as there were a few Ministries who colour coded there trackers manually and found it visually helpful. I also agree and have "RECORDS DUE' go into the negative once the due date has passed should be included

**Comment by arielleandrews** [comment]
@lmullane confirming the day the CFR is sent to the Ministry is not counted in the 10 days they have to respond. Example: CFR is sent on Sep 9, response is due Sep 23. 
@JHarrietha-AOT that makes good sense to me. Visual reminders are always helpful.

**Comment by lmullane** [comment]
Thanks, @arielleandrews. Is it the next business or the next calendar day?

**Comment by arielleandrews** [comment]
@lmullane next business day

**Comment by lmullane** [comment]
@arielleandrews, @JHarrietha-AOT, please review updated scenarios above reflecting your comments.

@JHarrietha-AOT, instead of negative days when overdue suggest we just state the number of days overdue.

**Comment by arielleandrews** [comment]
@lmullane Scenario 2 would follow the same principle and the clock/number of business days would update at midnight. We track always according to the number of days to the legislated due date and don't get into partial days

**Comment by JHarrietha-AOT** [comment]
@lmullane @arielleandrews Agree with Arielle that scenario 2 shouldn't be about the 24 hours passing but about it being past midnight and the next buisness day. Ans agree that days should show at 'X days past due' not in the negative.

Otherwise looks good.

**Comment by lmullane** [comment]
Displaying the CFR due date on the Request View page is not in scope for this story; it will be covered in a different story.
